### PR TITLE
Issue #10491 add HTMLInputElement attributes that reflect content ide…

### PR DESCRIPTION
**Pull request by amarant**
…ntically

it uses a new version of string-cache https://github.com/servo/string-cache/pull/148

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10505)

### CODE REVIEW

**Comment by Wafflespeanut** [comment]
./components/script/dom/htmlinputelement.rs:309: link to WHATWG may break in the future, use this format instead: https://html.spec.whatwg.org/multipage/#attr-input-accept

    ./components/script/dom/htmlinputelement.rs:312: link to WHATWG may break in the future, use this format instead: https://html.spec.whatwg.org/multipage/#attr-input-accept

    ./components/script/dom/htmlinputelement.rs:315: link to WHATWG may break in the future, use this format instead: https://html.spec.whatwg.org/multipage/#attr-input-alt

    ./components/script/dom/htmlinputelement.rs:317: trailing whitespace

    ./components/script/dom/htmlinputelement.rs:318: link to WHATWG may break in the future, use this format instead: https://html.spec.whatwg.org/multipage/#attr-input-alt

    ./components/script/dom/htmlinputelement.rs:321: link to WHATWG may break in the future, use this format instead: https://html.spec.whatwg.org/multipage/#attr-input-dirName

    ./components/script/dom/htmlinputelement.rs:323: trailing whitespace

    ./components/script/dom/htmlinputelement.rs:324: link to WHATWG may break in the future, use this format instead: https://html.spec.whatwg.org/multipage/#attr-input-dirName

**Comment by KiChjang** [comment]
I think those `#attr-input-*` links shoul really be `#dom-input-*` links instead.

**Comment by KiChjang** [comment]
Also, are the changes to `tests/wpt/metadata/html/dom/reflection-forms.html.ini` intended?

**Comment by jdm** [comment]
It's a test that setting properties of DOM APIs to certain values is properly reflected by the equivalent getter, so it seems likely to me.

**Comment by amarant** [comment]
I've fixed the urls, and tidy errors.

File: tests/wpt/mozilla/meta/mozilla/preferences.html.ini [hunk 1]
```diff
@@ -1,10 +1,3 @@
...
   type: testharness
-  prefs: [
-    dom.testbinding.preference_value.falsy:false,
-    dom.testbinding.preference_value.truthy:true,
-    dom.testbinding.preference_value.string_empty:,
-    dom.testbinding.preference_value.string_test:test,
-    dom.testbinding.preference_value.space_string_test:test1 test2,
-    dom.testbinding.preference_value.quote_string_test:"test1 test2",
-    ]
+  prefs: [dom.testbinding.preference_value.falsy:false, dom.testbinding.preference_value.truthy:true, dom.testbinding.preference_value.string_empty:, dom.testbinding.preference_value.string_test:test, dom.testbinding.preference_value.space_string_test:test1 test2, dom.testbinding.preference_value.quote_string_test:"test1 test2"]
```
Comment (KiChjang) [inline tests/wpt/mozilla/meta/mozilla/preferences.html.ini]: Why is this changed and set into one line?

File: tests/wpt/metadata/XMLHttpRequest/data-uri.htm.ini [hunk 2]
```diff
@@ -1,6 +1,5 @@
 [data-uri.htm]
   type: testharness
-
```
Comment (KiChjang) [inline tests/wpt/metadata/XMLHttpRequest/data-uri.htm.ini]: Not sure why this is a good change.

File: tests/wpt/mozilla/meta/mozilla/preferences.html.ini (hunk 1 above)
Comment (frewsxcv) [inline tests/wpt/mozilla/meta/mozilla/preferences.html.ini]: Running `./mach update-wpt` has some uncommitted changes that result in various formatting updates. This is one of them, though they should probably happen in a different PR.

Comment (amarant) [inline tests/wpt/mozilla/meta/mozilla/preferences.html.ini]: The thing is that I made the edit to format the ini file when I added the prefs in another PR, but it seems update-wpt format it in 1 line, maybe we should let it that way.

File: tests/wpt/metadata/XMLHttpRequest/data-uri.htm.ini (hunk 2 above)
Comment (amarant) [inline tests/wpt/metadata/XMLHttpRequest/data-uri.htm.ini]: This is the result of update-wpt, others doing the same command will have the same result, should I re-add the deleted line ?

Comment (frewsxcv) [inline tests/wpt/metadata/XMLHttpRequest/data-uri.htm.ini]: Personally, I think it's fine to leave it in. Ideally it'd be in a separate PR, but not a big deal at all. I'll let @KiChjang decide though

**Comment by KiChjang** [comment]
Forget about the changes, apparently update-wpt caused some formatting updates that I wasn't aware of. Thanks!

@bors-servo r+

**Comment by frewsxcv** [comment]
🎊 

```
Tests with unexpected results:
  ▶ Unexpected subtest result in /html/semantics/forms/the-input-element/datetime.html:
  └ PASS [expected FAIL] [date] The min attribute must have a value that is a valid global date and time string

  ▶ Unexpected subtest result in /html/semantics/forms/the-input-element/datetime.html:
  └ PASS [expected FAIL] [date] The max attribute must have a value that is a valid global date and time string

  ▶ Unexpected subtest result in /html/semantics/forms/the-input-element/datetime.html:
  └ PASS [expected FAIL] [date] The step attribute must be expressed in seconds

  ▶ Unexpected subtest result in /html/semantics/forms/the-input-element/datetime.html:
  └ PASS [expected FAIL] [time] The min attribute must have a value that is a valid global date and time string

  ▶ Unexpected subtest result in /html/semantics/forms/the-input-element/datetime.html:
  └ PASS [expected FAIL] [time] The max attribute must have a value that is a valid global date and time string

  ▶ Unexpected subtest result in /html/semantics/forms/the-input-element/datetime.html:
  └ PASS [expected FAIL] [time] The step attribute must be expressed in seconds

  ▶ Unexpected subtest result in /html/semantics/forms/the-input-element/datetime.html:
  └ PASS [expected FAIL] [datetime] The min attribute must have a value that is a valid global date and time string

  ▶ Unexpected subtest result in /html/semantics/forms/the-input-element/datetime.html:
  └ PASS [expected FAIL] [datetime] The max attribute must have a value that is a valid global date and time string

  ▶ Unexpected subtest result in /html/semantics/forms/the-input-element/datetime.html:
  └ PASS [expected FAIL] [datetime] The step attribute must be expressed in seconds

  ▶ Unexpected subtest result in /html/semantics/forms/the-input-element/datetime.html:
  └ PASS [expected FAIL] [month] The min attribute must have a value that is a valid global date and time string

  ▶ Unexpected subtest result in /html/semantics/forms/the-input-element/datetime.html:
  └ PASS [expected FAIL] [month] The max attribute must have a value that is a valid global date and time string

  ▶ Unexpected subtest result in /html/semantics/forms/the-input-element/datetime.html:
  └ PASS [expected FAIL] [month] The step attribute must be expressed in seconds

  ▶ Unexpected subtest result in /html/semantics/forms/the-input-element/datetime.html:
  └ PASS [expected FAIL] [week] The min attribute must have a value that is a valid global date and time string

  ▶ Unexpected subtest result in /html/semantics/forms/the-input-element/datetime.html:
  └ PASS [expected FAIL] [week] The max attribute must have a value that is a valid global date and time string

  ▶ Unexpected subtest result in /html/semantics/forms/the-input-element/datetime.html:
  └ PASS [expected FAIL] [week] The step attribute must be expressed in seconds

  ▶ Unexpected subtest result in /html/semantics/forms/the-input-element/email.html:
  └ PASS [expected FAIL] single_email doesn't have the multiple attribute

  ▶ Unexpected subtest result in /html/semantics/forms/the-input-element/email.html:
  └ PASS [expected FAIL] multiple_email has the multiple attribute

  ▶ Unexpected subtest result in /html/semantics/forms/the-input-element/range.html:
  └ PASS [expected FAIL] min attribute support on input element

  ▶ Unexpected subtest result in /html/semantics/forms/the-input-element/range.html:
  └ PASS [expected FAIL] max attribute support on input element

  ▶ Unexpected subtest result in /html/semantics/forms/the-input-element/time.html:
  └ PASS [expected FAIL] step attribute on default value check

  ▶ Unexpected subtest result in /html/semantics/forms/the-input-element/time.html:
  └ PASS [expected FAIL] max  attribute on default value check

  ▶ Unexpected subtest result in /html/semantics/forms/the-input-element/time.html:
  └ PASS [expected FAIL] min  attribute on default value check

  ▶ Unexpected subtest result in /html/semantics/forms/the-input-element/time.html:
  └ PASS [expected FAIL] max attribute support on input element

  ▶ Unexpected subtest result in /html/semantics/forms/the-input-element/time.html:
  └ PASS [expected FAIL] min attribute support on input element

  ▶ Unexpected subtest result in /html/semantics/forms/the-input-element/time.html:
  └ PASS [expected FAIL] step attribute support on input element
```

**Comment by amarant** [comment]
Ok I should have been more thorough during the wpt test, I will update the ini files. It means that Travis/AppVeyor don't do them ?

**Comment by frewsxcv** [comment]
Correct. Travis and Appveyor currently only run *unit* and *compile* tests, not *wpt* or an other tests:

https://github.com/servo/servo/blob/master/.travis.yml#L15-L16
